worlds/shivers/__init__.py

```diff
@@ -65,153 +71,184 @@ def create_regions(self) -> None:
+            library_location.place_locked_item(self.create_item("Key for Library"))
+            item_pool = [item for item in item_pool if item.name != "Key for Library"]
+        elif library_random == 3:
+            library_location.place_locked_item(self.create_item("Key for Three Floor Elevator"))
+            library_location_2 = self.random.choice(
+                [loc for loc in library_region.locations if
+                 not loc.name.startswith("Storage:") and loc != library_location]
+            )
+            library_location_2.place_locked_item(self.create_item("Key for Egypt Room"))
+            item_pool = [item for item in item_pool if
+                         item.name not in ["Key for Three Floor Elevator", "Key for Egypt Room"]]
+
+        # If front door option is on, determine which set of keys will
+        # be used for lobby access and add front door key to item pool
+        lobby_access_keys = 0
         if self.options.front_door_usable:
-            lobby_access_keys = self.random.randint(1, 2)
-            itempool += [self.create_item("Key for Front Door")]
+            lobby_access_keys = self.random.randint(0, 1)
+            item_pool.append(self.create_item("Key for Front Door"))
         else:
-            itempool += [self.create_item("Heal")]
-
-        self.multiworld.itempool += itempool
-
-        #Lobby acess:
-        if self.options.lobby_access == "early":
-            if lobby_access_keys == 1:
-                self.multiworld.early_items[self.player]["Key for Underground Lake Room"] = 1
-                self.multiworld.early_items[self.player]["Key for Office Elevator"] = 1
-                self.multiworld.early_items[self.player]["Key for Office"] = 1
-            elif lobby_access_keys == 2:
-                self.multiworld.early_items[self.player]["Key for Front Door"] = 1
-        if self.options.lobby_access == "local":
-            if lobby_access_keys == 1:
-                self.multiworld.local_early_items[self.player]["Key for Underground Lake Room"] = 1
-                self.multiworld.local_early_items[self.player]["Key for Office Elevator"] = 1
-                self.multiworld.local_early_items[self.player]["Key for Office"] = 1
-            elif lobby_access_keys == 2:
-                self.multiworld.local_early_items[self.player]["Key for Front Door"] = 1
-
-        #Pot piece shuffle location:
+            item_pool.append(self.create_item("Heal"))
+
+        def set_lobby_access_keys(items: Dict[str, int]):
+            if lobby_access_keys == 0:
+                items["Key for Underground Lake"] = 1
+                items["Key for Office Elevator"] = 1
+                items["Key for Office"] = 1
+            elif lobby_access_keys == 1:
+                items["Key for Front Door"] = 1
+
+        # Lobby access:
+        if self.options.lobby_access == 1:
+            set_lobby_access_keys(self.multiworld.early_items[self.player])
+        if self.options.lobby_access == 2:
```

Comment:
Should be able to use `elif`
```suggestion
        elif self.options.lobby_access == 2:
```